From ce7cf85fd47ec542b9ad82ee7aff7a4fa322ee29 Mon Sep 17 00:00:00 2001
From: Andy Fleming <afleming@freescale.com>
Date: Sun, 9 Aug 2009 13:00:35 -0500
Subject: [PATCH 204/252] p4080 debug: Change for debugging purposes

[Cleanly applied the FSL SDK 2.0.3 patch:
"kernel-2.6.30-0001-debug-Change-for-debugging-purposes.patch"
no Signed-off-by present in original patch]
Integrated-by: Yuri Nedel <Yuri.Nedel@windriver.com>
---
 arch/powerpc/kernel/traps.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/kernel/traps.c b/arch/powerpc/kernel/traps.c
index 3286e6d..96d6e89 100644
--- a/arch/powerpc/kernel/traps.c
+++ b/arch/powerpc/kernel/traps.c
@@ -320,7 +320,7 @@ static inline int check_io_access(struct pt_regs *regs)
 #ifndef CONFIG_FSL_BOOKE
 #define get_mc_reason(regs)	((regs)->dsisr)
 #else
-#define get_mc_reason(regs)	(mfspr(SPRN_MCSR) & MCSR_MASK)
+#define get_mc_reason(regs)	(mfspr(SPRN_MCSR))
 #endif
 #define REASON_FP		ESR_FP
 #define REASON_ILLEGAL		(ESR_PIL | ESR_PUO)
-- 
1.6.5.2

